### PR TITLE
[docs] Add KSS description to SGX introduction

### PR DESCRIPTION
**Pull request by DL8**
KSS stands for Key Separation and Sharing,
a feature added to the SGX hardware and
not yet implemented in Gramine

Signed-off-by: Lavi, Nir <nir.lavi@intel.com>

- - -
This change is [Reviewable](https://reviewable.io/reviews/gramineproject/gramine/873)

### CODE REVIEW

**Review by dimakuv — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/873)* status: all files reviewed, 6 unresolved discussions, not enough approvals from maintainers (3 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @DL8)

___
*[`-- commits` line 6 at r1](https://reviewable.io/reviews/gramineproject/gramine/873#-NAsUQJu4Ahh8uXsuwzz):*
The wrapping is weird. Git commit bodies are typically 72 chars per line. @dl8 No need for you to do anything, I'll fix this wrapping during final merge time.
___
*[`Documentation/sgx-intro.rst` line 357 at r1](https://reviewable.io/reviews/gramineproject/gramine/873#-NAsUiMtDUdHvGi8loHu:-NAsUiMtDUdHvGi8loHv:bepd7xq) ([raw file](https://github.com/gramineproject/gramine/blob/3288577f60f50e34555074fb576198133cd0265d/Documentation/sgx-intro.rst#L357)):*
> ```reStructuredText
>       attributes and configuration identifier. Extended enclave identity
>       is defined by the developer on enclave build, and configuration is
>       defined on enclave launch.
> ```

I would also add: `...on enclave launch and cannot be modified afterwards.`
___
*[`Documentation/sgx-intro.rst` line 360 at r1](https://reviewable.io/reviews/gramineproject/gramine/873#-NAsUsBUE1MSMH2ZNtHD:-NAsUsBUE1MSMH2ZNtHE:b-bj1ofy) ([raw file](https://github.com/gramineproject/gramine/blob/3288577f60f50e34555074fb576198133cd0265d/Documentation/sgx-intro.rst#L360)):*
> ```reStructuredText
> 
>       In addition to the calculated enclave and signer measurements, developer
>       is expected to define a product ID and :term:`SVN` for its enclaves.
> ```

`its` -> `her` (we typically use `her` when we talk about people, e.g. `developer`, `user`; but definitely not `its`)
___
*[`Documentation/sgx-intro.rst` line 368 at r1](https://reviewable.io/reviews/gramineproject/gramine/873#-NAsVDM2-DqFelAbBKbi:-NAsVDM2-DqFelAbBKbj:b-etdfad) ([raw file](https://github.com/gramineproject/gramine/blob/3288577f60f50e34555074fb576198133cd0265d/Documentation/sgx-intro.rst#L368)):*
> ```reStructuredText
>       enclave launch. These attributes are part of :term:`SGX Report` and
>       additional key policy attributes are added to provide fine-grained
>       control over key derivation.
> ```

I would split this sentence into two: one about SGX report, another about key derivation.
___
*[`Documentation/sgx-intro.rst` line 370 at r1](https://reviewable.io/reviews/gramineproject/gramine/873#-NAsVRnoC4dlwhMDGfqu:-NAsVRnoC4dlwhMDGfqv:b-ev6rin) ([raw file](https://github.com/gramineproject/gramine/blob/3288577f60f50e34555074fb576198133cd0265d/Documentation/sgx-intro.rst#L370)):*
> ```reStructuredText
>       control over key derivation.
> 
>       New build attributes:
> ```

You need to add an empty line in-between the title of the list and the list items. This is a requirement of RestructredText (RST) format that we use for documentation. (See examples in the very beginning of this file.)
___
*[`Documentation/sgx-intro.rst` line 374 at r1](https://reviewable.io/reviews/gramineproject/gramine/873#-NAsVbgG9ldLhNkgm_al:-NAsVbgG9ldLhNkgm_am:b-ev6rin) ([raw file](https://github.com/gramineproject/gramine/blob/3288577f60f50e34555074fb576198133cd0265d/Documentation/sgx-intro.rst#L374)):*
> ```reStructuredText
>       - Family ID
> 
>       New enclave launch attributes:
> ```

You need to add an empty line in-between the title of the list and the list items. This is a requirement of RestructredText (RST) format that we use for documentation. (See examples in the very beginning of this file.)

**Review by DL8 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/873)* status: 0 of 1 files reviewed, 6 unresolved discussions, not enough approvals from maintainers (3 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @dimakuv and @DL8)

___
*[`Documentation/sgx-intro.rst` line 357 at r1](https://reviewable.io/reviews/gramineproject/gramine/873#-NAsUiMtDUdHvGi8loHu:-NAsiznxDus8WXFVOAe5:b-896fix) ([raw file](https://github.com/gramineproject/gramine/blob/3288577f60f50e34555074fb576198133cd0265d/Documentation/sgx-intro.rst#L357)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

I would also add: `...on enclave launch and cannot be modified afterwards.`
</blockquote></details>

Done.
___
*[`Documentation/sgx-intro.rst` line 360 at r1](https://reviewable.io/reviews/gramineproject/gramine/873#-NAsUsBUE1MSMH2ZNtHD:-NAsj2E79RQPxnKFkprz:b-896fix) ([raw file](https://github.com/gramineproject/gramine/blob/3288577f60f50e34555074fb576198133cd0265d/Documentation/sgx-intro.rst#L360)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

`its` -> `her` (we typically use `her` when we talk about people, e.g. `developer`, `user`; but definitely not `its`)
</blockquote></details>

Done.
___
*[`Documentation/sgx-intro.rst` line 368 at r1](https://reviewable.io/reviews/gramineproject/gramine/873#-NAsVDM2-DqFelAbBKbi:-NAsiympDo0h9_8djED6:b-896fix) ([raw file](https://github.com/gramineproject/gramine/blob/3288577f60f50e34555074fb576198133cd0265d/Documentation/sgx-intro.rst#L368)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

I would split this sentence into two: one about SGX report, another about key derivation.
</blockquote></details>

Done.
___
*[`Documentation/sgx-intro.rst` line 370 at r1](https://reviewable.io/reviews/gramineproject/gramine/873#-NAsVRnoC4dlwhMDGfqu:-NAsix8ZDoYS5mvDZ24w:b-896fix) ([raw file](https://github.com/gramineproject/gramine/blob/3288577f60f50e34555074fb576198133cd0265d/Documentation/sgx-intro.rst#L370)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

You need to add an empty line in-between the title of the list and the list items. This is a requirement of RestructredText (RST) format that we use for documentation. (See examples in the very beginning of this file.)
</blockquote></details>

Done.
___
*[`Documentation/sgx-intro.rst` line 374 at r1](https://reviewable.io/reviews/gramineproject/gramine/873#-NAsVbgG9ldLhNkgm_al:-NAsikq60P13qRFkL9g6:b-896fix) ([raw file](https://github.com/gramineproject/gramine/blob/3288577f60f50e34555074fb576198133cd0265d/Documentation/sgx-intro.rst#L374)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

You need to add an empty line in-between the title of the list and the list items. This is a requirement of RestructredText (RST) format that we use for documentation. (See examples in the very beginning of this file.)
</blockquote></details>

Done.

**Review by dimakuv — APPROVED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/873)* status: all files reviewed, 3 unresolved discussions, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: Intel) (waiting on @DL8)

*[a discussion](https://reviewable.io/reviews/gramineproject/gramine/873#-NAsnDo35QtZe0UW2evX:-NAsnDo35QtZe0UW2evY:bc28a3r) (no related file):*
I'm Ok with this PR now. Needs at least one more review, and then can be merged. Thanks @dl8!

___
___
*[`-- commits` line 11 at r2](https://reviewable.io/reviews/gramineproject/gramine/873#-NAsmuNn9Gxl0BJHvVBW):*
You should use the "git fixup" workflow when submitting updates to the original commit. Please see https://gramine.readthedocs.io/en/latest/devel/contributing.html#pr-life-cycle

Also, you can google "git fixup flow" and read how it works. There is nothing too complex about it.

(No need to change anything now, just for the future.)

**Review by DL8 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/873)* status: all files reviewed, 3 unresolved discussions, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: Intel) (waiting on @dimakuv)

*[a discussion](https://reviewable.io/reviews/gramineproject/gramine/873#-NAsnDo35QtZe0UW2evX:-NBH-PcRELnpwbWsG66z:b32wtd) (no related file):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

I'm Ok with this PR now. Needs at least one more review, and then can be merged. Thanks @dl8!
</blockquote></details>

ok

___
___
*[`-- commits` line 6 at r1](https://reviewable.io/reviews/gramineproject/gramine/873#-NAsUQJu4Ahh8uXsuwzz):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

The wrapping is weird. Git commit bodies are typically 72 chars per line. @dl8 No need for you to do anything, I'll fix this wrapping during final merge time.
</blockquote></details>

ok
___
*[`-- commits` line 11 at r2](https://reviewable.io/reviews/gramineproject/gramine/873#-NAsmuNn9Gxl0BJHvVBW):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

You should use the "git fixup" workflow when submitting updates to the original commit. Please see https://gramine.readthedocs.io/en/latest/devel/contributing.html#pr-life-cycle

Also, you can google "git fixup flow" and read how it works. There is nothing too complex about it.

(No need to change anything now, just for the future.)
</blockquote></details>

ok

**Review by mkow — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/873)* status: all files reviewed, 8 unresolved discussions, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: Intel) (waiting on @dimakuv and @DL8)

___
*[`-- commits` line 2 at r2](https://reviewable.io/reviews/gramineproject/gramine/873#-NBvc5t6Ed4Z--drJqKw):*
(see https://gramine.readthedocs.io/en/latest/devel/contributing.html)



_Suggestion:_
```
[Docs]
```
___
*[`-- commits` line 6 at r2](https://reviewable.io/reviews/gramineproject/gramine/873#-NBvcIhY0SsjsqCxNy9e):*
> and not yet implemented in Gramine

This implies that we plan to implement it, which is not true (we haven't decided yet). Please change to "and currently not implemented in Gramine".
___
*[`-- commits` line 8 at r2](https://reviewable.io/reviews/gramineproject/gramine/873#-NBvca7e9aF0HwK3zcQP):*
Why have you swapped your name and surname? Looks weird :)



_Code quote:_
```
Lavi, Nir <nir.lavi@intel.com>
```
___
*[`Documentation/sgx-intro.rst` line 363 at r2](https://reviewable.io/reviews/gramineproject/gramine/873#-NBvda728GtThm8-5QJN:-NBvda728GtThm8-5QJO:undefined) ([raw file](https://github.com/gramineproject/gramine/blob/ec73a6e1f526eea2ac6471eec5d8f9bc00081ff9/Documentation/sgx-intro.rst#L363)):*
> ```reStructuredText
>       These identifiers are part of the :term:`SGX Report` and are expected to
>       be used in :term:`Attestation`. They are also used by SGX key derivation
>       to derive different keys per enclave.
> ```





_Suggestion:_
```reStructuredText
different keys per configuration
```
___
*[`Documentation/sgx-intro.rst` line 367 at r2](https://reviewable.io/reviews/gramineproject/gramine/873#-NBvdh9UACIlFx-K9OXW:-NBvdh9UACIlFx-K9OXX:bm0wn7g) ([raw file](https://github.com/gramineproject/gramine/blob/ec73a6e1f526eea2ac6471eec5d8f9bc00081ff9/Documentation/sgx-intro.rst#L367)):*
> ```reStructuredText
>       KSS adds two more attributes for enclave build and two new ones for
>       enclave launch, which are part of the :term:`SGX Report`.
>       Additionally,  key policy attributes are extended to provide fine-grained
> ```

double space

**Review by DL8 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/873)* status: 0 of 1 files reviewed, 8 unresolved discussions, not enough approvals from maintainers (2 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @dimakuv and @mkow)

___
*[`-- commits` line 2 at r2](https://reviewable.io/reviews/gramineproject/gramine/873#-NBvc5t6Ed4Z--drJqKw):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

(see https://gramine.readthedocs.io/en/latest/devel/contributing.html)
</blockquote></details>

will fix
___
*[`-- commits` line 6 at r2](https://reviewable.io/reviews/gramineproject/gramine/873#-NBvcIhY0SsjsqCxNy9e):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

> and not yet implemented in Gramine

This implies that we plan to implement it, which is not true (we haven't decided yet). Please change to "and currently not implemented in Gramine".
</blockquote></details>

will fix
___
*[`-- commits` line 8 at r2](https://reviewable.io/reviews/gramineproject/gramine/873#-NBvca7e9aF0HwK3zcQP):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

Why have you swapped your name and surname? Looks weird :)
</blockquote></details>

this is how my name is presented in org emails, so i use the same format to be consistent. it is critical?
___
*[`Documentation/sgx-intro.rst` line 363 at r2](https://reviewable.io/reviews/gramineproject/gramine/873#-NBvda728GtThm8-5QJN:-NCFu41l7dHfv5SESVh1:bgvs6h8) ([raw file](https://github.com/gramineproject/gramine/blob/ec73a6e1f526eea2ac6471eec5d8f9bc00081ff9/Documentation/sgx-intro.rst#L363)):*
> ```reStructuredText
>       These identifiers are part of the :term:`SGX Report` and are expected to
>       be used in :term:`Attestation`. They are also used by SGX key derivation
>       to derive different keys per enclave.
> ```

will fix
___
*[`Documentation/sgx-intro.rst` line 367 at r2](https://reviewable.io/reviews/gramineproject/gramine/873#-NBvdh9UACIlFx-K9OXW:-NCFu57P7ocar86YHVrg:bgvs6h8) ([raw file](https://github.com/gramineproject/gramine/blob/ec73a6e1f526eea2ac6471eec5d8f9bc00081ff9/Documentation/sgx-intro.rst#L367)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

double space
</blockquote></details>

will fix

**Review by dimakuv — DISMISSED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/873)* status: all files reviewed, 6 unresolved discussions, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: Intel) (waiting on @mkow)

___
*[`-- commits` line 2 at r2](https://reviewable.io/reviews/gramineproject/gramine/873#-NBvc5t6Ed4Z--drJqKw):*
<details><summary><i>Previously, DL8 (Nir) wrote…</i></summary><blockquote>

will fix
</blockquote></details>

@dl8 You're supposed to write "Done" once you've fixed. This forces Reviewable to automatically mark this comment as resolved. Just for the future.
___
*[`-- commits` line 8 at r2](https://reviewable.io/reviews/gramineproject/gramine/873#-NBvca7e9aF0HwK3zcQP):*
<details><summary><i>Previously, DL8 (Nir) wrote…</i></summary><blockquote>

this is how my name is presented in org emails, so i use the same format to be consistent. it is critical?
</blockquote></details>

No, it's not critical. @mkow This is the Intel way, no idea why (other than easier alphabetic sorting).

**Comment by dimakuv** [comment]
Jenkins, test this please

**Review by mkow — DISMISSED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/873)* status: all files reviewed, 1 unresolved discussion (waiting on @DL8)

___
*[`-- commits` line 2 at r2](https://reviewable.io/reviews/gramineproject/gramine/873#-NBvc5t6Ed4Z--drJqKw):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

@dl8 You're supposed to write "Done" once you've fixed. This forces Reviewable to automatically mark this comment as resolved. Just for the future.
</blockquote></details>

Not necessarily, you can write anything, just mark the conversation as resolved from your side ;)
___
*[`-- commits` line 8 at r2](https://reviewable.io/reviews/gramineproject/gramine/873#-NBvca7e9aF0HwK3zcQP):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

No, it's not critical. @mkow This is the Intel way, no idea why (other than easier alphabetic sorting).
</blockquote></details>

But this is an open source project, not an Intel-internal one ;) So, you're actually being inconsistent this way ;) Anyways, I won't block on this.
___
*[`-- commits` line 5 at r3](https://reviewable.io/reviews/gramineproject/gramine/873#-NCV9IzUAZdRgF5uozXX):*
We can fix this during rebase.



_Suggestion:_
```
implemented in Gramine.
```

**Review by mkow — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/873)* status: all files reviewed, all discussions resolved, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: ITL)

**Review by dimakuv — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/873)* status: :shipit: complete! all files reviewed, all discussions resolved

**Comment by dimakuv** [comment]
Jenkins, test this plesae

**Comment by dimakuv** [comment]
Jenkins, test this please

**Pull request merged**
